jetson/percep/test_protocol.md

```diff
@@ -0,0 +1,73 @@
+## Compile Test
+- [ ] Offline Debug: ` ./jarvis build jetson/percep -o with_zed=false ar_detection=true obs_detection=true perception_debug=true `
+  - [ ] No compiler errors or warnings
+- [ ] Offline Silent: ` ./jarvis build jetson/percep -o with_zed=false ar_detection=true obs_detection=true perception_debug=false `
+  - [ ] No compiler errors or warnings
+- [ ] ZED Debug: ` ./jarvis build jetson/percep -o with_zed=true ar_detection=true obs_detection=true perception_debug=true `
+  - [ ] No compiler errors, only ZED warnings
+- [ ] ZED Silent: ` ./jarvis build jetson/percep -o with_zed=true ar_detection=true obs_detection=true perception_debug=false `
+   - [ ] No compiler errors, only ZED warnings
+
+## Offline Debug Test
+- [ ] `./jarvis build jetson/percep -o with_zed=false ar_detection=true obs_detection=true perception_debug=true`
+- [ ] Run code against test1 image set located in perception-files repository
+- Check for:
+   - [ ]  Left and right path are being calculated
+   - [ ]  Calculated bearing is reasonable given viewer output
+   - [ ]  AR Tag ID is 2
+   - [ ]  Obstacles with large width are detected
+   - [ ]  Path is projected around not through large obstacles
+   - [ ]  Short obstacles are detected
+   - [ ]  Steep inclines are detected as obstacles
+- [ ] Run code against test2 image set located in perception-files repository
+- Check for:
+   - [ ]  Left and right path are being calculated
+   - [ ]  Calculated bearing is reasonable given viewer output
+   - [ ]  AR Tag ID is 6
+   - [ ]  2 AR Tags detected simultaneously
+   - [ ]  Obstacles with large width are detected
+   - [ ]  Path is projected around not through large obstacles
+   - [ ]  Short obstacles are detected
+   - [ ]  Steep inclines are detected as obstacles
+
+## Offline Silent Test
+- [ ] `./jarvis build jetson/percep -o with_zed=false ar_detection=true obs_detection=true perception_debug=false`
+- [ ] Run `time ./jarvis exec jetson/percep` against test_case1 image set
+- [ ] Run `time ./jarvis exec jetson/percep` against test_case2 image set
+- [ ] User value is less than or equal to \<TODO Determine Value\>
+
+## Unit Test
+- [ ] ` ./jarvis build jetson/percep -o with_zed=true ar_detection=true obs_detection=true perception_debug=true `
+- [ ] `./jarvis exec lcm_bridge/server/`
+- [ ] `./jarvis exec lcm_tools_echo Obstacle "/obstacle"`
```

Comment:
Would piping the output to a log file make it easier for testers to look through the output?

Reply:
I think it would be more helpful for them to just slow the program down, which we are able to do. Since the unit test won't always be the exact same as other unit tests, we can't compare the log files, and it will be hard for us to line up the log file output with the camera input.